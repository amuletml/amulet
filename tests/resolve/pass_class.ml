class semigroup 'a
  val (<>) : 'a -> 'a -> 'a

instance semigroup int
  let (<>) = (+)
