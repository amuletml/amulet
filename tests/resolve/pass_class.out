class () => semigroup#0 a#1  begin
  val <>#2 : 'a#1 -> 'a#1 -> 'a#1
end
instance () => semigroup#0 int#-2 begin
  <>#2 = ((+#-6))
end
class () => default#3 a#4  begin
  val a#5 : 'a#4
  val b#6 : 'a#4
  let a#5 = b#6
end
